src/main/java/teammates/ui/controller/InstructorEditInstructorFeedbackPageAction.java

```diff
@@ -0,0 +1,105 @@
+package teammates.ui.controller;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import teammates.common.datatransfer.FeedbackParticipantType;
+import teammates.common.datatransfer.FeedbackQuestionAttributes;
+import teammates.common.datatransfer.FeedbackResponseAttributes;
+import teammates.common.datatransfer.FeedbackSessionQuestionsBundle;
+import teammates.common.datatransfer.InstructorAttributes;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.util.Assumption;
+import teammates.common.util.Const;
+import teammates.logic.api.GateKeeper;
+
+/**
+ * The {@code InstructorEditInstructorFeedbackPageAction} class handles incoming requests to the page.
+ * {@code FeedbackSubmissionEditPageData} will be generated and the page requested will be loaded.
+ */
+public class InstructorEditInstructorFeedbackPageAction extends Action {
+
+    @Override
+    protected ActionResult execute() throws EntityDoesNotExistException {
+        String courseId = getRequestParamValue(Const.ParamsNames.COURSE_ID); 
+        String feedbackSessionName = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
+        String moderatedEntityIdentifier = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_MODERATED_STUDENT);
+        String moderatedQuestionNumber = getRequestParamValue("moderatedquestion");
+
+        Assumption.assertNotNull(String.format(Const.StatusMessages.NULL_POST_PARAMETER_MESSAGE, 
+                                               Const.ParamsNames.COURSE_ID), 
+                                 courseId);
+        Assumption.assertNotNull(String.format(Const.StatusMessages.NULL_POST_PARAMETER_MESSAGE, 
+                                               Const.ParamsNames.FEEDBACK_SESSION_NAME), 
+                                 feedbackSessionName);
+        Assumption.assertNotNull(String.format(Const.StatusMessages.NULL_POST_PARAMETER_MESSAGE, 
+                                               Const.ParamsNames.FEEDBACK_SESSION_MODERATED_STUDENT), 
+                                 moderatedEntityIdentifier);
+
+        new GateKeeper().verifyAccessible(
+                logic.getInstructorForGoogleId(courseId, account.googleId), 
+                logic.getFeedbackSession(feedbackSessionName, courseId),
+                false, Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION);
+        
+        InstructorAttributes instructorUnderModeration = logic.getInstructorForEmail(courseId, moderatedEntityIdentifier);
+
+        // If the instructor doesn't exist
+        if (instructorUnderModeration == null) {
+            throw new EntityDoesNotExistException("Instructor Email "
+                    + moderatedEntityIdentifier + " does not exist in " + courseId
+                    + ".");
+        }
+
+        FeedbackSubmissionEditPageData data = new FeedbackSubmissionEditPageData(account, student);
```

Comment:
The student variable comes from the parent Action.java
I think it is used to track unregistered students  @LowWeiLin